alonet/panoptic/panoptic.py

```diff
@@ -0,0 +1,217 @@
+                - "pred_boxes": The normalized boxes coordinates for all queries, represented as
+                                (center_x, center_y, height, width). These values are normalized in [0, 1],
+                                relative to the size of each individual image (disregarding possible padding).
+                                See PostProcess for information on how to retrieve the unnormalized bounding box.
+                - "pred_masks": Binary masks, each one to assign to predicted boxes.
+                                Shape = [batch_size x num_queries x H // 4 x W // 4]
+                - "bb_outputs": Backbone outputs, requered in this forward
+                - "enc_outputs": Transformer encoder outputs, requered on this forward
+                - "dec_outputs": Transformer decoder outputs, requered on this forward
+                - "infer_info": Parameters to use in inference procedure
+                - "aux_outputs": Optional, only returned when auxilary losses are activated. It is a list of
+                                dictionnaries containing the two above keys for each decoder layer.
+        """
+        # DETR model forward to obtain box embeddings
+        out = self.detr(frames, **kwargs)
+        features = out["bb_outputs"]
+        src, mask = features[-1][0], features[-1][1].to(torch.float32)
+        mask = mask[:, 0].to(torch.bool)
+        bs = src.shape[0]
+
+        # Filter boxes and get mask indices from them
+        get_filter_fn = get_filter_fn or (lambda *args, **kwargs: get_mask_queries(*args, model=self.detr, **kwargs))
+        dec_outputs, filters, gmq_params = get_filter_fn(frames=frames, m_outputs=out, **kwargs)
+
+        # FIXME h_boxes takes the last one computed, keep this in mind
+        # Use box embeddings as input of Multi Head attention
+        bbox_mask = self.bbox_attention(dec_outputs, out["enc_outputs"], mask=mask)
+        # And then, use MHA ouput as input of FPN-style CNN
+        seg_masks = self.mask_head(self.detr.input_proj(src), bbox_mask, [features[i][0] for i in range(3)[::-1]])
+
+        out["pred_masks"] = seg_masks.view(bs, bbox_mask.shape[1], seg_masks.shape[-2], seg_masks.shape[-1])
+        out["infer_info"] = {"frame_size": frames.shape[-2:], "gmq_params": gmq_params, "filters": filters}
+        return out  # Return the DETR output + pred_masks
+
+    @torch.no_grad()
+    def inference(self, forward_out: Dict, maskth: int = 0.5, **kwargs):
+        """Given the model forward outputs, this method will return an aloscene.Mask tensor with binary
+        mask with its corresponding label per object detected.
+
+        Parameters
+        ----------
+        forward_out: dict
+            Dict with the model forward outptus
+
+        Returns
+        -------
+        aloscene.BoundingBoxes2D
+            Boxes from DETR model
+        aloscene.Maks
+            Binary mask of PanopticHead
+        """
+
+        # Get information about boxes
```

Comment:
would it be possible to surcharge the inference method of Detr instead of doing the boxes inference again ?

Reply:
@thibo73800   Before, I made this procedure. But, In term of align boxes and masks, I decided to make the boxes "again". The reasons:
* The inference need the b_filters to align the masks.
To make sure that the filters will be the same, the inference create the boxes. But if you don't agree, I can try it.

Reply:
I might be wrong but I think that both the  inference of deformable and detr can use a filter in the infernece method
detr.inference(filter=my_filter). If the filter is defined, then the model will use it instead of using the default one. Let me know

Reply:
Ok I will try it. Thanks